Dockerfile

```diff
@@ -1,33 +1,48 @@
-FROM phusion/baseimage:0.9.18
+FROM phusion/baseimage:0.9.19
 ENV DEBIAN_FRONTEND noninteractive
 
-RUN echo "debconf shared/accepted-oracle-license-v1-1 select true" | debconf-set-selections
-RUN echo "debconf shared/accepted-oracle-license-v1-1 seen true" | debconf-set-selections
-
 # First, install add-apt-repository and bzip2
-RUN apt-get update
-RUN apt-get -y install software-properties-common python-software-properties bzip2 unzip openssh-client git lib32stdc++6 lib32z1
-
 # Add oracle-jdk8 to repositories
-RUN add-apt-repository ppa:webupd8team/java
-
-# Update apt
-RUN apt-get update
-
 # Install oracle-jdk8
-RUN apt-get -y install oracle-java8-installer
+# Install Ruby
+RUN echo "debconf shared/accepted-oracle-license-v1-1 select true" | debconf-set-selections && \
+    echo "debconf shared/accepted-oracle-license-v1-1 seen true" | debconf-set-selections && \
+    apt-get update && \
+    apt-get -y install software-properties-common python-software-properties curl bzip2 unzip openssh-client git lib32stdc++6 lib32z1 && \
+    add-apt-repository ppa:webupd8team/java && \
+    apt-add-repository ppa:brightbox/ruby-ng && \
+    apt-get update && \
+    apt-get -y install oracle-java8-installer ruby2.3 ruby2.3-dev build-essential && \
+    gem install bundler && \
+    rm -rf /var/cache/oracle-jdk8-installer && \
+    apt-get -y remove software-properties-common python-software-properties && \
+    apt-get -y autoremove && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
 
 # Install android sdk
-RUN wget http://dl.google.com/android/android-sdk_r24.4.1-linux.tgz
-RUN tar -xvzf android-sdk_r24.4.1-linux.tgz
-RUN mv android-sdk-linux /usr/local/android-sdk
-RUN rm android-sdk_r24.4.1-linux.tgz
+RUN curl -sf -o android-sdk_r24.4.1-linux.tgz -L http://dl.google.com/android/android-sdk_r24.4.1-linux.tgz && \
+    tar -xzf android-sdk_r24.4.1-linux.tgz && \
+    mv android-sdk-linux /usr/local/android-sdk && \
+    rm android-sdk_r24.4.1-linux.tgz
```

Comment:
i'm wondering can we make `android-sdk_r24.4.1-linux.tgz` a variable, so that we only need to update version number once